enterprise/api_client/enterprise.py

```diff
@@ -0,0 +1,50 @@
+
+# -*- coding: utf-8 -*-
+"""
+Client for communicating with the Enterprise API.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.conf import settings
+from django.core.cache import cache
+
+from enterprise.api_client.lms import LmsApiClient
+from enterprise.utils import get_cache_key, traverse_pagination
+
+
+class EnterpriseApiClient(LmsApiClient):
+    """
+    Object builds an API client to make calls to the Enterprise API.
+    """
+
+    API_BASE_URL = settings.LMS_ROOT_URL + 'enterprise/api/v1/'
+
+    def get_all_enterprise_courses(self, enterprise_customer):
+        """
+        Query the Enter API for the course details of the given course_id.
+
+        Arguments:
+            enterprise_customer (Enterprise Customer): Enterprise customer for fetching courses
+
+        Returns:
+            dict: A dictionary containing details about the course, in an enrollment context (allowed modes, etc.)
+        """
+        api_resource_name = 'enterprise-customer'
```

Comment:
@brittneyexline @zubair-arbi I think Brittney is right; see https://github.com/edx/edx-enterprise/blob/master/tests/test_enterprise/api/test_views.py#L30 which uses that endpoint